core/src/main/java/org/frankframework/filesystem/FileSystemActor.java

```diff
@@ -542,26 +535,20 @@ private String arrangeFolder(String determinedFolderName) throws FileSystemExcep
 		return determinedFolderName;
 	}
 
-	private void writeContentsToFile(OutputStream out, Message input, ParameterValueList pvl) throws IOException, FileSystemException {
-		Object contents;
-		if (pvl!=null && pvl.contains(PARAMETER_CONTENTS1)) {
-			contents=pvl.get(PARAMETER_CONTENTS1).getValue();
+	private InputStream getContents(Message input, ParameterValueList pvl, String charset) throws IOException {
+		final Message message;
+		if (pvl != null && pvl.contains(PARAMETER_CONTENTS1)) {
+			message = pvl.get(PARAMETER_CONTENTS1).asMessage();
 		} else {
-			contents=input;
+			message = input;
 		}
-		out = augmentOutputStream(out);
-		if (contents instanceof Message) {
-			StreamUtil.streamToStream(((Message)contents).asInputStream(), out);
-		} else if (contents instanceof InputStream) {
-			StreamUtil.streamToStream((InputStream)contents, out);
-		} else if (contents instanceof byte[]) {
-			out.write((byte[])contents);
-		} else if (contents instanceof String) {
-			out.write(((String) contents).getBytes(StringUtils.isNotEmpty(getCharset()) ? getCharset() : StreamUtil.DEFAULT_INPUT_STREAM_ENCODING));
-		} else {
-			throw new FileSystemException("expected Message, InputStream, ByteArray or String but got [" + contents.getClass().getName() + "] instead");
+
+		try(InputStream is = message.asInputStream(charset)) {
```

Comment:
Ik denk niet dat je hier een try-with-resources moet gebruiken omdat je de `is` terug geeft.

Dat gaat in unit-tests misschien goed als de inputstream iets is als een ByteArrayInputStream die leest van memory en nooit echt gesloten wordt maar ik verwacht dat dit "in het echt" fout zal gaan.

Reply:
Zal ik testen!